osu.Game/Overlays/Toolbar/ToolbarRulesetSelector.cs

```diff
@@ -1,49 +1,51 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-using System.Linq;
 using osu.Framework.Allocation;
-using osu.Framework.Bindables;
 using osu.Framework.Caching;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Effects;
 using osuTK;
-using osuTK.Input;
 using osuTK.Graphics;
 using osu.Framework.Graphics.Shapes;
-using osu.Framework.Input.Events;
 using osu.Game.Rulesets;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Bindables;
+using osu.Framework.Input.Events;
+using osuTK.Input;
+using System.Linq;
 
 namespace osu.Game.Overlays.Toolbar
 {
-    public class ToolbarRulesetSelector : Container
+    public class ToolbarRulesetSelector : TabControl<RulesetInfo>
```

Comment:
It'd be good to split this out into an abstract class, `RulesetSelector`. It would contain none of the visual elements but it would implement the bindable / bindable disabling / ruleset store -related stuff.

`ToolbarRulesetSelector` would then implement it to add the visuals as in here. What I'd like to see achieved is using it to replace the custom implementation inside `DirectOverlay`'s `FilterControl`.